pysocialforce/sim_view.py

```diff
@@ -145,25 +168,78 @@ def _resize_window(self):
             (self.width, self.height), pygame.RESIZABLE)
         self.screen.blit(old_surface, (0, 0))
 
-    def _zoom_camera(self, state: VisualizableSimState) \
+    def _scale_pedestrian_state(self, state: VisualizableSimState) \
             -> Tuple[VisualizableSimState, Tuple[float, float]]:
         state.pedestrian_positions *= self.scaling
         state.ped_actions *= self.scaling
-        return state, (0, 0)
+        return state
 
     def _draw_pedestrians(self, ped_pos: np.ndarray):
         for ped_x, ped_y in ped_pos:
-            pygame.draw.circle(self.screen, PED_COLOR, (ped_x, ped_y), self.ped_radius * self.scaling)
+            pygame.draw.circle(
+                self.screen,
+                PED_COLOR,
+                (ped_x+self.offset[0], ped_y+self.offset[1]),
+                self.ped_radius * self.scaling
+                )
 
-    def _draw_obstacles(self, offset: Tuple[float, float]):
-        offset = offset[0], offset[1]
-        self.screen.blit(self.surface_obstacles, offset)
+    def _draw_obstacles(self):
+        self.screen.blit(self.surface_obstacles, self.offset)
 
     def _augment_ped_actions(self, ped_actions: np.ndarray):
+        """Draw the actions of the pedestrians as lines."""
         for p1, p2 in ped_actions:
-            pygame.draw.line(self.screen, PED_ACTION_COLOR, p1, p2, width=3)
-
-    def _augment_timestep(self, timestep: int):
-        text = f'step: {timestep}'
-        text_surface = self.font.render(text, False, TEXT_COLOR)
-        self.screen.blit(text_surface, self.timestep_text_pos)
+            pygame.draw.line(
+                self.screen,
+                PED_ACTION_COLOR,
+                p1+self.offset,
+                p2+self.offset,
+                width=3
+                )
+
+    def _add_text(self, timestep: int):
+        text_lines = [
+            f'step: {timestep}',
+            f'offset: {self.offset[0]}, {self.offset[1]}',
+            f'scaling: {self.scaling}'
```

Comment:
change order and apply scaling to offset

Reply:
has been resolved in https://github.com/ll7/pysocialforce-ll7/pull/1/commits/af90e2d6c4d8dabd29fc7ba62b191dd2c4e0643b